media-gfx/openscad/openscad-2021.01.ebuild

```diff
@@ -0,0 +1,105 @@
+# Copyright 1999-2021 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+inherit elisp-common qmake-utils xdg
+
+SITEFILE="50${PN}-gentoo.el"
+
+DESCRIPTION="The Programmers Solid 3D CAD Modeller"
+HOMEPAGE="https://www.openscad.org/"
+SRC_URI="https://github.com/${PN}/${PN}/releases/download/${P}/${P}.src.tar.gz -> ${P}.tar.gz"
+
+LICENSE="GPL-2"
+SLOT="0"
+KEYWORDS="~amd64 ~arm64 ~x86"
+IUSE="ccache emacs"
```

Comment:
Do we really need a USE=ccache? Is FEATURES=ccache broken?

Reply:
Don't use it, so I don't know whether it's broken. It's to avoid an automagic dep and use if the binary is found.
See https://github.com/gentoo/gentoo/pull/19714/files#diff-b734e3660f655b6e0ef09b77e8ebe066d32ce4cb5edd7f38a95a5ea7934d3d05

Reply:
Will take some time to test this out on a clean machine. The package has lots of dependencies.

Reply:
What I mean is the patch can probably be unconditionally applied. If the build respects CC/CXX then FEATURES=ccache should do everything necessary to use it - explicit build system support shouldn't be necessary.

Reply:
I see.
FEATURES=ccache won't pull in the ccache package though, so isn't the USE flag necessary to pull the package in if users have set the feature?

Reply:
It's working if I leave the USE flag to pull in the package.

Reply:
If a user wants FEATURES=ccache they can install it themselves. Ebuilds don't need a dependency on it.

Reply:
Note that this is sometimes misleading because upstreams expose a ccache option but we usually want to force it off. When we're using ccache with Portage, they don't need to know about it.

Reply:
> If a user wants FEATURES=ccache they can install it themselves. Ebuilds don't need a dependency on it.

Understood. Although it's IMO counter-intuitive. If I, as a user, add ccache to FEATURES, I expect the package manager and / or the ebuild to pull in any necessary dependencies, so the package in question can immediately use it. Like with FEATURES="test", which activates a USE flag and pulls in any needed deps.

> Note that this is sometimes misleading because upstreams expose a ccache option but we usually want to force it off. When we're using ccache with Portage, they don't need to know about it.

Thanks for the concise explanation!